### PR TITLE
Update spring-boot version to 2.6.0

### PR DESCRIPTION
**Pull request by DanielFran**
Fix https://github.com/jhipster/generator-jhipster/issues/16772

Depends on https://github.com/jhipster/jhipster-bom/pull/500
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by mraible** [comment]
It looks like we might not be able to upgrade to Spring Boot 2.6.x if we want to continue supporting Springfox for Swagger. 

https://github.com/springfox/springfox/issues/3934

What do you think we should do @jhipster/developers?

**Comment by atomfrede** [comment]
Spring doc looks fine. If I remember correctly we quite some issues with springfox before. So let's drop it and use spring doc I would say.

**Comment by pascalgrimaud** [comment]
I can confirm: springdoc works well, we should migrate.
Can someone open a new dedicated ticket for that?

**Comment by mraible** [comment]
> I can confirm: springdoc works well, we should migrate. Can someone open a new dedicated ticket for that?

Done. https://github.com/jhipster/generator-jhipster/issues/17175

**Comment by mraible** [comment]
R2DBC apps are failing with:

```
FROM bank_account e LEFT OUTER JOIN jhi_user e_user ON e.user_id = e_user.id WHERE id = 3 WHERE e.id = 3]; nested 
exception is io.r2dbc.spi.R2dbcBadGrammarException: [90059] [90059] Ambiguous column name "ID"; SQL statement:
```

Anyone know how to fix?

**Comment by mraible** [comment]
@ecostanzi Since you helped migrate our R2DBC implementation to Spring Boot 2.4, maybe you can assist with the upgrade to Spring Boot 2.6 too? I'm seeing a strange error about [ambiguous column names](https://github.com/jhipster/generator-jhipster/pull/16787#issuecomment-989420629).

**Comment by mraible** [comment]
I think the R2DBC SQL error might be caused by this block of code:

```java
String select = entityManager.createSelect(selectFrom, Operation.class, pageable, criteria);
String alias = entityTable.getReferenceName().getReference();
String selectWhere = Optional
    .ofNullable(criteria)
    .map(crit ->
        new StringBuilder(select)
            .append(" ")
            .append("WHERE")
            .append(" ")
            .append(alias)
            .append(".")
            .append(crit)
            .toString()
    )
    .orElse(select); // TODO remove once https://github.com/spring-projects/spring-data-jdbc/issues/907 will be fixed
return db.sql(selectWhere).map(this::process);
```

@Tcharl Since you entered the GitHub issue in the comment above, do you know how to fix it?

**Comment by Tcharl** [comment]
Hmmm, I don't think the error comes from here, but I can explain why I sent an issue to spring: Basically, it was impossible to add a constant in a SQL query due to the absence of a condition letting us to declare constants (so impossible to declare 'toto' in a `where name='Toto'` statement).
I guess that it's now solved so we should be able to improve the above code (I'll take a look after my cucumber PR if it's not done since).

Also, I hope that now we can extends https://github.com/spring-projects/spring-data-r2dbc/blob/main/src/main/java/org/springframework/data/r2dbc/repository/support/SimpleR2dbcRepository.java instead of do it all by ourselved (only find with eager should be implemented manually I guess)

**Comment by mraible** [comment]
@Tcharl It'd be great if you could look into the R2DBC issue. Please create a new issue to track. That way I can add a bug bounty to it.

**Comment by mraible** [comment]
@Tcharl I created an [issue for R2DBC with Spring Boot 2.6](https://github.com/jhipster/generator-jhipster/issues/17271) and added a bounty to it.

**Comment by Tcharl** [comment]
Don't you think we can do something with testcontainers instead this embedded mongo which is a real pain for us?

**Comment by Tcharl** [comment]
Works flawlessly on my machine, but ends-up in CI: anyone already experienced cpu/ram issue with reactive?

**Comment by mshima** [comment]
@Tcharl my guess would be that mongodb reactive is conflicting with imperative:
https://github.com/jhipster/generator-jhipster/blob/9f95694e8ec478b716bb89f54e153b162c85a19f/generators/server/templates/pom.xml.ejs#L541-L546

There is a mongock 5 probably we should migrate to it.

**Comment by Tcharl** [comment]
OK to try: I've no other bullet... still, the actual setup works well on local and most of the tests are passing in ci... I'm trying memory and cpu tweaks without any success for now.

**Comment by DanielFran** [comment]
@Tcharl updated reference for the mongock bom

**Comment by Tcharl** [comment]
@mshima Note that V5 is not compatible with reactive either! we'll have to provide our own driver if we want to handle reactive

**Comment by mshima** [comment]
@Tcharl thanks for the information, I saw https://mongock.io/v4/reactive/index.html for version 4.
Could not find similar information for version 5.

**Comment by Tcharl** [comment]
From tests on my side: the [mongodb-springdata-v3-driver driver](https://github.com/mongock/mongock/blob/develop/drivers/mongock-driver-mongodb/mongodb-springdata-v3-driver/src/main/java/io/mongock/driver/mongodb/springdata/v3/config/SpringDataMongoV3Context.java) need a MongoTemplate (non-reactive).

Going reactive would mean to create a `mongodb-springdata-v3-reactive-driver`, plus an `mongodb-async-v4-driver` I guess.

**Comment by Tcharl** [comment]
Last idea I have in mind, but you won't like it...
Binding mongodb migration a new 'db-migration' maven-profile (can be renamed can drop no-liquibase after that).
Then deactivating this profile in backend tests (thus everything related to data migration).

**Comment by dieppa** [comment]
> From tests on my side: the [mongodb-springdata-v3-driver driver](https://github.com/mongock/mongock/blob/develop/drivers/mongock-driver-mongodb/mongodb-springdata-v3-driver/src/main/java/io/mongock/driver/mongodb/springdata/v3/config/SpringDataMongoV3Context.java) need a MongoTemplate (non-reactive).
> 
> Going reactive would mean to create a `mongodb-springdata-v3-reactive-driver`, plus an `mongodb-async-v4-driver` I guess.

Hello, one of the Mongock's members here.

Although its use has been extended(cli, etc.) the main use of Mongock is to be executed synchronously in the application startup, normally before it's deployed and even aborting the deployment in case of failure. That's way we haven't really paid attention to reactive drivers.

However, as @Tcharl mentioned, it would require to add a driver for it. **We are happy to help on this**

**Comment by Tcharl** [comment]
Still folks, we have a long way to go 'till we get mongo reactive working:
Look at what I discovered removing the spring-data-mongodb (non-reactive) driver:
```
[ERROR] Failures: 
[ERROR]   AccountResourceIT.testGetExistingAccount:141 No matching value at JSON path "$.authorities" //this one is normal
[ERROR]   AccountResourceIT.testRegisterAdminIsIgnored:478  //this one is normal
Expected size: 1 but was: 0 in:
[]
[ERROR]   DivisionResourceIT.getAllDivisions:262 Status expected:<200 OK> but was:<501 NOT_IMPLEMENTED>
[ERROR]   DivisionResourceIT.getAllDivisionsAsStream:230 Status expected:<200 OK> but was:<501 NOT_IMPLEMENTED>
[ERROR]   DivisionResourceIT.getDivision:294 Status expected:<200 OK> but was:<501 NOT_IMPLEMENTED>
[ERROR]   PlaceResourceIT.getAllPlaces:247 Status expected:<200 OK> but was:<501 NOT_IMPLEMENTED>
[ERROR]   PlaceResourceIT.getAllPlacesAsStream:215 Status expected:<200 OK> but was:<501 NOT_IMPLEMENTED>
[ERROR]   PlaceResourceIT.getPlace:297 Status expected:<200 OK> but was:<501 NOT_IMPLEMENTED>
[ERROR]   PublicUserResourceIT.getAllAuthorities:95 No values for JSON path "$[?(@=='ROLE_ADMIN')]" //this one is normal
[ERROR] Errors: 
[ERROR]   DivisionResourceIT.createDivision » UnsupportedOperation DBRef resolution is n...
[ERROR]   DivisionResourceIT.deleteDivision » UnsupportedOperation DBRef resolution is n...
[ERROR]   DivisionResourceIT.fullUpdateDivisionWithPatch » UnsupportedOperation DBRef re...
[ERROR]   DivisionResourceIT.partialUpdateDivisionWithPatch » UnsupportedOperation DBRef...
[ERROR]   DivisionResourceIT.putNewDivision » UnsupportedOperation DBRef resolution is n...
[ERROR]   PlaceResourceIT.createPlace » UnsupportedOperation DBRef resolution is not sup...
[ERROR]   PlaceResourceIT.deletePlace » UnsupportedOperation DBRef resolution is not sup...
[ERROR]   PlaceResourceIT.fullUpdatePlaceWithPatch » UnsupportedOperation DBRef resoluti...
[ERROR]   PlaceResourceIT.partialUpdatePlaceWithPatch » UnsupportedOperation DBRef resol...
[ERROR]   PlaceResourceIT.putNewPlace » UnsupportedOperation DBRef resolution is not sup...

```

**Comment by dieppa** [comment]
@Tcharl I see you are quite very busy with this, let us know if it will help if Mongock team provides the Reactive driver.

We are also bussy, so it would be nice to let us know when this is required.

Thanks

**Comment by Tcharl** [comment]
Hi @dieppa,

For sure it will help us, so that it will be compatible with the stack!
In parallel, I (and maybe another committer I hope) will make the entire DSL working well in a reactive environment

**Comment by dieppa** [comment]
Great, we'll do that  😉 . 

Can you tell me the timing for this?   We need to organise it other priorities.

Thanks

**Comment by Tcharl** [comment]
Whenever you want, it's the beauty of OSS! we'll adapt in the meantime: it's not a full blocker as we can always introduce blocking driver to achieve the thing...

**Comment by Tcharl** [comment]
@mshima I just disabled the blocking driver in ci backend tests (and was forced to disable some tests relative to data import too) but the issue is still there... IMHO, this is definitively a github runner performance issue (only [2 cores](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners)), and it would be the absolute same with Azure devops which [has the same](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml#hardware).

So two options here:
- Either I'll continue the try and die approach tweaking the container's configuration spec here and there.
- Buying a Microsoft sub for the community to host an AzDo Hosted runner for our resource intensive test: I'm also skilled in terraform and Ansible ;-)

File: generators/server/files.js [hunk 1]
```diff
@@ -1707,6 +1737,25 @@ const baseServerFiles = {
         },
       ],
     },
+    {
```
Comment (Tcharl) [inline generators/server/files.js:1740]: Remove these two duplicates

File: generators/server/templates/src/main/java/package/config/dbmigrations/InitialSetupMigration.java.ejs [hunk 2]
```diff
@@ -76,11 +91,29 @@ public class InitialSetupMigration {
         userUser.setLastName("User");
         userUser.setEmail("user@localhost");
         userUser.setActivated(true);
-        userUser.setLangKey("<%= nativeLanguage %>");
+        userUser.setLangKey("en");
```
Comment (Tcharl) [inline generators/server/templates/src/main/java/package/config/dbmigrations/InitialSetupMigration.java.ejs:94]: Rollback

**Comment by mshima** [comment]
Tested mongodb locally with this branch and it worked (backend test and cypress).
I cannot see the GitHub resources as too low for these use cases, I think it's a bug somewhere else.

let's try to release v7.5.0 with 2.5.8.
And since mongodb+reative is broken at ci only, lets move them to daily builds.

What do you think @pascalgrimaud.

**Comment by Tcharl** [comment]
Of course it works, but only on our machine :-p.
I don't know how to 'move tests to daily builds', can you please explain or do?

Thank you @mshima , I would have been lost without your help

**Comment by pascalgrimaud** [comment]
Ok I can prepare a new release for the new year :)

**Comment by mshima** [comment]
I did lots of checks to see if I managed to find the reason of the failures and tried to workaround them.
Those failures look related to lack of synchronicity with mongodb database.
Our tests reuses primary keys, which looks to make the database request to lock.

I've disabled the backend tests failures, so CI will succeed even if the backend tests fails.

I started working on the fix, but it's far from trivial.
We have 3 choices:
- Merge and wait if upstream fixes the deadlock.
- Merge and implement the fix.
- Keep working on the fix at the branch.

Just created https://github.com/jhipster/generator-jhipster/issues/17478 with the proposed fix.

**Comment by pascalgrimaud** [comment]
My opinion:

We should go ahead and merge it.
We put too much effort on mongodb + reactive, which is not used by our end users.

**Comment by mshima** [comment]
I agree about the effort on mongodb + reactive.

But IMO backend test should be improved.
Those deleteAll at BeforeEach are bad practices:
https://github.com/jhipster/generator-jhipster/blob/829bfc59d29d8042c2da52796baf92521d41da6a/generators/server/templates/src/test/java/package/web/rest/UserResourceIT.java.ejs#L271-L276

And we may enable parallel testing.

**Comment by mshima** [comment]
All green.
Anything to change @DanielFran @Tcharl ?

**Comment by pascalgrimaud** [comment]
Good news.
I'll have time this friday to release JH v7.6.0

**Comment by DanielFran** [comment]
Good, so let's merge

**Pull request merged**